### ISSUE TITLE
assembly more fragmented than trusted contigs

### ISSUE THREAD

**Issue by TC-Hewitt** [ablab/spades#579]
hello,

I ran an assembly with spades 3.14.1 (see below command) using trusted contigs from a prior polished assembly combined with short reads from a new sequencing run. My thinking was that with the added depth and coverage of the short reads Spades may be able to close some gaps of the original assembly and generate longer contigs. However, the Spades output seems to be more fragmented and shorter than the original assembly of trusted contigs (see image). The number of contigs >20kb is significantly less. Spades appears to have fragmented the trusted contigs. Could you explain why this might be? The short reads are from the same biological sample the prior assembly was made from.

spades.py -1 reads_R1.fq -2 reads_R2.fq -s reads_SE.fq --trusted-contigs prior_assembly.fa -o Spades_out
![contig_counts](https://user-images.githubusercontent.com/33470968/94101017-5be9d780-fe72-11ea-8b92-74da983078ef.png)

**Comment by asl** [comment]
Hello

Thank you for your interest in SPAdes
>  My thinking was that with the added depth and coverage of the short reads Spades may be able to close some gaps of the original assembly and generate longer contigs. 

This is a very common error. It works in opposite way (as documented in SPAdes manual): trusted contigs are used to supplement repeat resolution, close coverage gaps, etc. in the assembly of new data. There is no way to "upgrade" the existing assembly and in case of trusted contigs all structural differences, etc. between the contigs and new reads are resolved in favor of new reads.

So, long story short: you are assembling new data and SPAdes will provide you assembly of new data. When there is an artifact of new data (e.g. coverage gap) that could be closed using trusted contigs, then an assembler will do it. However, in all contradictory cases new assembly wins.

We understand that this is not what you're wanting, but you're essentially wanting reference-guided assembly which SPAdes does not support. Still, we are working on improving trusted contigs mode. So, you may expect some changes in the next releases.

**Comment by TC-Hewitt** [comment]
Oh I see, thanks for clarifying. Yes it seems to be a common misunderstanding as several forum posts state that Spades works in this way. I will try a reference-guided alternative for now. Thanks.

**Comment by asl** [comment]
Good luck! And beware of misassemblies!
